pkg/types/router_context.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2024 The Aibrix Team.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/vllm-project/aibrix/pkg/utils"
+	v1 "k8s.io/api/core/v1"
+)
+
+const podMetricPort = "8000"
+
+type RequestFeatures []float64
+
+// RoutingAlgorithms defines the routing algorithms
+type RoutingAlgorithms string
+
+// RoutingContext encapsulates the context information required for routing.
+// It can be extended with more fields as needed in the future.
+type RoutingContext struct {
+	context.Context
+	Algorithm   RoutingAlgorithms
+	Model       string
+	Message     string
+	RequestTime time.Time
+	PendingLoad float64
+
+	chTargetPod chan *v1.Pod
+	targetPod   *v1.Pod
+	tokens      []int
+}
+
+var requestPool = sync.Pool{
+	New: func() any {
+		return &RoutingContext{
+			chTargetPod: make(chan *v1.Pod),
+		}
+	},
+}
+
+func (alg RoutingAlgorithms) NewContext(ctx context.Context, model string, message string) *RoutingContext {
+	request := requestPool.Get().(*RoutingContext)
+	request.reset(ctx, alg, model, message)
+	return request
+}
+
+func NewRoutingContext(ctx context.Context, algorithms RoutingAlgorithms, model string, message string) *RoutingContext {
+	request := requestPool.Get().(*RoutingContext)
+	request.reset(ctx, algorithms, model, message)
+	return request
+}
+
+func (r *RoutingContext) Delete() {
+	requestPool.Put(r)
+}
+
+func (r *RoutingContext) PromptTokens() ([]int, error) {
```

Comment:
Remove this function for now or return input text as string.

Reason is that utils.TokenizeInputText is legacy method which will be removed and implementations will move to tokenizer interface which returns bytearray.